packages/ember-data/lib/system/store.js

```diff
@@ -1522,15 +1522,53 @@ Store = Service.extend({
   /**
     Push some data for a given type into the store.
 
-    This method expects normalized data:
+    This method expects [JSON API](http://jsonapi.org/) compound document with [your primary data wrapped inside `data` property](http://jsonapi.org/format/#document-top-level):
 
-    * The ID is a key named `id` (an ID is mandatory)
-    * The names of attributes are the ones you used in
-      your model's `DS.attr`s.
-    * Your relationships must be:
-      * represented as IDs or Arrays of IDs
-      * represented as model instances
-      * represented as URLs, under the `links` key
+    ```
+    store.push({
+      data: {
+        // primary data for single record of type `Person`
+        id: '1',
+        type: 'person',
+        attributes: {
+          firstName: 'Daniel'
+        }
+      }
+    })
+    ```
+
+    `data` property can also hold an array (of records):
+
+    ```
+    store.push({
+      data: [
+        // an array of records
+        {
+          id: '1',
+          type: 'person',
+          attributes: {
+            firstName: 'Daniel'
+          }
+        },
+        {
+          id: '2',
+          type: 'person',
+          attributes: {
+            firstName: 'Tom'
+          }
+        }
+      ]
+    })
+    ```
+
+    There are some typical properties for `JSONAPI` payload:
+    * `id` - mandatory, unique record's key
+    * `type` - mandatory string which matches `model` name (can be singular or plural form)
+    * `attributes` - object which holds data for record attributes - `DS.attr`'s declared in model
+    * `relationships` - object which must contain any of the following:
```

Comment:
Could you mention that `links`, `data` and `meta` are supposed to be under each relationships' respective key, to remove confusion.